src/components/WeatherWidget/WeatherWidget.tsx

```diff
@@ -86,76 +64,24 @@ export default function WeatherWidget({ location }: Props) {
     fetchForecast();
   }, [location, selectedLanguage, selectedTemperatureUnit, enqueueSnackbar]);
 
-  const formatTemperatureUnits = (tempUnit: TemperatureUnits) => {
-    switch (tempUnit) {
-      case TemperatureUnits.KELVIN:
-        return '\u00B0K';
-      case TemperatureUnits.CELSIUS:
-        return '\u2103';
-      case TemperatureUnits.FAHRENHEIT:
-        return '\u00B0F';
-      default:
-        throw new Error('New temperature unit found');
-    }
-  };
-  const formatTemperatureData = (temp: number, tempUnit: TemperatureUnits) =>
-    `${Math.floor(temp)}${formatTemperatureUnits(tempUnit)}`;
-
   return isLoading ? (
     <CircularProgress size={SPINNER_SIZE} />
   ) : (
-    <table>
-      <caption>
-        {location &&
-          `${findCountryNameByCode(location.country)}, ${location.name}`}
-      </caption>
-      <thead>
-        <tr>
-          <th>Date/Hours</th>
-          {HOURS.map((hour) => (
-            <th key={hour}>{hour}</th>
-          ))}
-        </tr>
-      </thead>
-      <tbody>
-        {forecast?.map(([date, weather], index) => {
-          const emptyCells = TIME_PERIODS_AMOUNT - weather.length;
-          return (
-            <tr key={date}>
-              <td>{date}</td>
-              {index === 0 &&
-                Array.from({ length: emptyCells }).map((_, idx) => (
-                  // We should leave some cells empty because the weather API doesn't provide data for the past hours of the current day. Also, some cells at the end of the table are empty because data is only provided for the next 5 days
-                  <td key={idx}> </td>
-                ))}
-              {weather.map((weatherReport) => {
-                const {
-                  dt,
-                  weather: [{ icon, description }],
-                  main: { temp, humidity },
-                } = weatherReport;
-                return (
-                  <td key={dt}>
-                    <Tooltip title={description}>
-                      <img
-                        src={`${
-                          import.meta.env.VITE_BASE_URL
-                        }img/wn/${icon}.png`}
-                        alt='Weather condition'
-                      />
-                    </Tooltip>
-                    {`Temperature: ${formatTemperatureData(
-                      temp,
-                      selectedTemperatureUnit,
-                    )}
-                    Humidity: ${humidity}%`}
-                  </td>
-                );
-              })}
-            </tr>
-          );
-        })}
-      </tbody>
-    </table>
+    <Box>
+      {forecast && (
```

Comment:
1. Don't duplicate this check. Also convert `forecast` to boolean explicitly
2. Don't render `Box` if there is no content